packages/SwingSet/src/kernel/liveSlots.js

```diff
@@ -623,9 +659,20 @@ function build(
     assert(!didRoot);
     didRoot = true;
 
+    const disavowPowers = {};
+    if (enableDisavow) {
+      disavowPowers.disavow = disavow;
+    }
+
     // here we finally invoke the vat code, and get back the root object
     const rootObject = buildRootObject(
-      harden({ D, exitVat, exitVatWithFailure, ...vatPowers }),
+      harden({
+        D,
+        exitVat,
+        exitVatWithFailure,
+        ...disavowPowers,
+        ...vatPowers,
+      }),
```

Comment:
This seems like a weird idiom for injecting the `disavow` power into the vat powers.  Why not just have `disavow` be a function or `undefined` and skip the object spread?

Reply:
From what I could tell, `disavow = undefined; vp = { D, exitVat, disavow }; Object.keys(vp)` makes it look like `disavow` is still present, even though the value is `undefined`, and I figured that disabling `disavow` should make it go away completely. I agree this is a weird idiom. I suppose another approach would be to build up a variable (named something different), conditionally add a `.disavow` property, and then harden it:

```js
const vp = { D, exitVat, exitVatWithFailure, ...vatPowers };
if (enableDisavow) {
  vp.disavow = disavow;
}
const rootObject = buildRootObject(harden(vp), harden(vatParameters));
```

(one tiny difference, if `disavow` also appeared in the incoming `vatPowers`, which one takes precedence.. I don't think it matters at all)

Would that pattern seem better? It'd probably be more consise, given `prettier`'s enthusiasm for whitespace..